File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int compute_index(int letters, int words, int sentences);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    printf("l: %i\n", letters);
    int words = count_words(text);
    printf("w: %i\n", words);
    int sentences = count_sentences(text);
    printf("s: %i\n", sentences);
    int index = compute_index(letters, words, sentences);

    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }

    if (index > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }

    printf("Grade %i\n", index);
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }

    return ++count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char c = text[i];
        if (c == '.' || c == '?' || c == '!')
        {
            count++;
        }
    }

    return count;
}

/** Calculate Coleman-Liau index */
int compute_index(int letters, int words, int sentences)
{
    double L = (letters * 100) / (double) words;
    double S = (sentences * 100) / (double) words;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    return round(index);
}

File: sentimental-readability/readability.py
import cs50
import sys


def main():
    text = cs50.get_string("Text: ")
    letters = count_letters(text)
    # print(f"l: {letters}")
    words = count_words(text)
    # print(f"w: {words}")
    sentences = count_sentences(text)
    # print(f"s: {sentences}")
    index = compute_index(letters, words, sentences)

    if (index < 1):
        print("Before Grade 1")
        sys.exit(0)

    if (index > 16):
        print("Grade 16+")
        sys.exit(0)

    print(f"Grade {index}")


def count_letters(text):
    count = 0
    for i in range(len(text)):
        if text[i].isalpha():
            count += 1
    return count


def count_words(text):
    count = 0
    for i in range(len(text)):
        c = text[i]
        if c == " ":
            count += 1
    count += 1
    return count


def count_sentences(text):
    count = 0
    for i in range(len(text)):
        c = text[i]
        if c == "." or c == "?" or c == "!":
            count += 1
    return count


def compute_index(letters, words, sentences):
    """Calculate Coleman-Liau index"""

    L = (letters * 100) / words
    S = (sentences * 100) / words
    index = 0.0588 * L - 0.296 * S - 15.8
    return round(index)


if __name__ == "__main__":
    main()